File: news.py
from random import randrange
day_by_day = []
all_day = []
rn_ton = 0
ton = 0
rn_demand = 0

demand = 0
revenue = 0
lp = 0
slvg = 0
purchase_value_of_70_np = round((.33 * 70), 2)
dp = 0

def rn_for_ton(k): #2 function
    for i in range (k):
        global rn_ton, day_by_day
        day_by_day = [] # reset the list to empty
        rn_ton = randrange(100)
        day_by_day.append(i+1)
        day_by_day.append(rn_ton)
        type_of_news_day() #3 call

def type_of_news_day(): #3 function
    global ton
    if rn_ton < 36:
        ton = "good"
    elif rn_ton < 81:
        ton = "fair"
    elif rn_ton < 101:
        ton = "poor"
    day_by_day.append(ton)
    
    rn_for_demand() #4 call

def rn_for_demand(): #4 function
    global rn_demand
    rn_demand = randrange(100)
    day_by_day.append(rn_demand)
    _demand() #5 call
    
def _demand(): #5 function
    global demand
    if ton == "poor" and rn_demand < 45:
        demand = 40
    elif ton == "poor" and rn_demand < 67:
        demand = 50
    elif ton == "poor" and rn_demand < 83:
        demand = 60
    elif ton == "poor" and rn_demand < 95:
        demand = 70
    elif ton == "poor" and rn_demand < 101:
        demand = 80
    elif ton == "fair" and rn_demand < 11:
        demand = 40
    elif ton == "fair" and rn_demand < 29:
        demand = 50
    elif ton == "fair" and rn_demand < 69:
        demand = 60
    elif ton == "fair" and rn_demand < 89:
        demand = 70
    elif ton == "fair" and rn_demand < 97:
        demand = 80
    elif ton == "fair" and rn_demand < 101:
        demand = 90
    elif ton == "good" and rn_demand < 4:
        demand = 40
    elif ton == "good" and rn_demand < 9:
        demand = 50
    elif ton == "good" and rn_demand < 24:
        demand = 60
    elif ton == "good" and rn_demand < 44:
        demand = 70
    elif ton == "good" and rn_demand < 79:
        demand = 80
    elif ton == "good" and rn_demand < 94:
        demand = 90
    elif ton == "good" and rn_demand < 101:
        demand = 100
    else:
        demand = 0
    
    day_by_day.append(demand)
    _revenue() #6 call
    
def _revenue(): #6 function
    global revenue
    revenue = demand * .5
    day_by_day.append(revenue)
    profit_lost() #7 call
    
def profit_lost(): #7 function
    global lp
    if demand > 70:
        lp = (demand - 70) * .17
    else:
        lp = 0
    day_by_day.append(round(lp, 2))
    salvage() #8 call

def salvage(): #8 function
    global slvg
    if demand < 70:
        slvg = (70 - demand) * .05
    else:
        slvg = 0
    day_by_day.append(slvg)
    daily_profit() #9 call

def daily_profit(): #9 function
    global dp
    dp = revenue - purchase_value_of_70_np - lp + slvg
    day_by_day.append(round(dp, 2))
    day_by_day.append(purchase_value_of_70_np)
    all_day.append(day_by_day)
    #end of one loop initiated in function 2
    
rn_for_ton(30) #1 #2 call

import pandas as pd
df = pd.DataFrame(all_day, columns = ['Day', ' RN for TON ', ' TON ', ' RN for Demand ', ' Demand ', ' Revenue ', ' Lost Profit ', ' Salvage ', ' Daily Profit ', ' Purchase Value of 70'])

df=df.set_index('Day')
pd.set_option('display.colheader_justify', 'center')

print(df)

File: news.c
#include<stdio.h>
#include <stdlib.h>
#include <time.h>

int i, rn_ton , ton, rn_demand, demand, revenue;
float lp, salvage, dp, purchase_value_of_70_np = .33 * 70;

void rn_for_ton(int k){
    for(i = 0; i < k; i++){
        
        rn_ton = rand()%100;
        type_of_news_day();
    }
}

void type_of_news_day(){
    if(rn_ton < 36){
        ton = "good";
    }
    else if(rn_ton < 81){
        ton = "fair";
    }
    else if(rn_ton < 101){
        ton = "poor";
    }
    rn_for_demand();
}

void rn_for_demand(){
    rn_demand = rand()%100;
    _demand();
}

void _demand(){
    if(ton == "poor" & rn_demand < 45){
        demand = 40;
    }
    else if(ton == "poor" & rn_demand < 67){
        demand = 50;
    }
    else if(ton == "poor" & rn_demand < 83){
        demand = 60;
    }
    else if(ton == "poor" & rn_demand < 95){
        demand = 70;
    }
    else if(ton == "poor" & rn_demand < 101){
        demand = 80;
    }
    else if(ton == "fair" & rn_demand < 11){
        demand = 40;
    }
    else if(ton == "fair" & rn_demand < 29){
        demand = 50;
    }
    else if(ton == "fair" & rn_demand < 69){
        demand = 60;
    }
    else if(ton == "fair" & rn_demand < 89){
        demand = 70;
    }
    else if(ton == "fair" & rn_demand < 97){
        demand = 80;
    }
    else if(ton == "fair" & rn_demand < 101){
        demand = 90;
    }
    else if(ton == "good" & rn_demand < 4){
        demand = 40;
    }
    else if(ton == "good" & rn_demand < 9){
        demand = 50;
    }
    else if(ton == "good" & rn_demand < 24){
        demand = 60;
    }
    else if(ton == "good" & rn_demand < 44){
        demand = 70;
    }
    else if(ton == "good" & rn_demand < 79){
        demand = 80;
    }
    else if(ton == "good" & rn_demand < 94){
        demand = 90;
    }
    else if(ton == "good" & rn_demand < 101){
        demand = 100;
    }
    _revenue();
}

void _revenue(){
    revenue = demand * .5;
    _lost_profit();
}

void _lost_profit(){
    if(demand > 70){
        lp = (demand - 70) * .17;
    }
    else{
        lp = 0;
    }
    _salvage();
}

void _salvage(){
    if(demand < 70){
        salvage = (70 - demand) * .05;
    }
    else{
        salvage = 0;
    }
    _daily_profit();
}

void _daily_profit(){
    dp = revenue - purchase_value_of_70_np - lp + salvage;
    printf("%d\t   %d\t   %s\t    %d\t\t   %d\t   %d\t\t %.1f\t   %.1f\t\t       %.1f\t      %.1f\t\n", i+1, rn_ton, ton, rn_demand, demand, revenue, lp, salvage, dp, purchase_value_of_70_np);
}

int main(){
    printf("DAY\t Rn_TON\t   TON\t RN_DEMAND\t DEMAND\t REVENUE    LOST PROFIT\t SALVAGE\t DAILY PROFIT\t  Purchase Value of 70\t\n");
    srand(time(0));
    rn_for_ton(30);
    return 0;
}